frontend/server/src/main/java/com/amazonaws/ml/mms/metrics/MetricCollector.java

```diff
@@ -86,15 +85,77 @@ public String collect() throws IOException {
         return IOUtils.toString(p.getInputStream(), StandardCharsets.UTF_8);
     }
 
+    public String getWorkerInfo() {
+        ModelManager modelManager = ModelManager.getInstance();
+        List<WorkerThread> workers = modelManager.getWorkers();
+        List<WorkerEntry> entries = new ArrayList<>();
+        for (WorkerThread worker : workers) {
+            entries.add(new WorkerEntry(worker.getPid(), worker.getGpuId()));
+        }
+        if (entries.isEmpty()) {
+            return "[]";
+        }
+        return JsonUtils.GSON.toJson(entries);
+    }
+
+    public String collectSystemLevelMetrics() throws IOException {
+        String[] args = new String[2];
+        args[0] = "python";
+        args[1] = "mms/metrics/system_metrics.py";
+        // run the "python script to collectSystemLevelMetrics metrics" command
+        // using the Runtime exec method:
+        return collect(args);
+    }
+
+    public String collectProcessMemoryMetrics() throws IOException {
+        String[] args = new String[3];
+        args[0] = "python";
+        args[1] = "mms/metrics/process_memory_metric.py";
+        args[2] = getWorkerInfo();
```

Comment:
command line may have length limit, consider use stdin to pass to python process